test/e2e-go/restAPI/restClient_test.go

```diff
@@ -186,6 +187,36 @@ func waitForTransaction(t *testing.T, testClient libgoal.Client, fromAddress, tx
 	}
 }
 
+func waitForTransactionV2(t *testing.T, testClient libgoal.Client, fromAddress, txID string, timeout time.Duration) (tx v1.Transaction, err error) {
+	a := require.New(fixtures.SynchronizedTest(t))
+	rnd, err := testClient.Status()
+	a.NoError(err)
+	if rnd.LastRound == 0 {
+		t.Fatal("it is currently round 0 but we need to wait for a transaction that might happen this round but we'll never know if that happens because ConfirmedRound==0 is indestinguishable from not having happened")
+	}
+	timeoutTime := time.Now().Add(timeout * time.Second)
```

Comment:
This should not multiply by `time.Second` since `timeout` is already a `time.Duration`